frame/system/src/lib.rs

```diff
@@ -1110,6 +1126,7 @@ impl<T: Trait> Module<T> {
 	/// To be called immediately after finishing the initialization of the block
 	/// (e.g., called `on_initialize` for all modules).
 	pub fn note_finished_initialize() {
+		Self::update_accumulators();
```

Comment:
Why do we need to call this here?

Reply:
Because of the changes trie (cc @svyatonik). It needs to be aware of what changed within particular transaction, thus we need to update storage during both block construction and execution.

Reply:
Is the changes trie really working on transaction level? Always thought that this is only per block.

Reply:
Had the same impression, but consulted with @svyatonik